File: apps/codemap-search/tests/fixtures/navigation/java/nested.java
import java.util.List;
import org.springframework.stereotype.Service;

@Service
class CheckoutService {
    private final OrderRepository repository;
    private final Inventory inventory;
    private final OrderMapper mapper;

    CheckoutService(OrderRepository repository, Inventory inventory, OrderMapper mapper) {
        this.repository = repository;
        this.inventory = inventory;
        this.mapper = mapper;
    }

    @Transactional
    Receipt submit(List<OrderLine> lines) {
        OrderDto dto = mapper.map(lines);
        AuditEvent event = AuditEvent.audit("checkout.submit", dto.total());

        inventory.reserve(lines);
        repository.save(dto);
        event.commit();
        return Receipt.from(dto);
    }

    static final class Receipt {
        static Receipt from(OrderDto dto) {
            return new Receipt();
        }
    }
}

File: apps/codemap-search/tests/fixtures/navigation/kotlin/nested.kt
import com.example.OrderMapper
import org.springframework.stereotype.Service

@Service
class CheckoutService(
    private val repository: OrderRepository,
    private val inventory: Inventory,
    private val mapper: OrderMapper,
) {
    @Transactional
    fun submit(lines: List<OrderLine>): Receipt {
        val dto: OrderDto = mapper.map(lines)
        val event = AuditEvent.audit("checkout.submit", dto.total)

        inventory.reserve(lines)
        repository.save(dto)
        event.commit()
        return Receipt.from(dto)
    }

    companion object {
        fun empty(): Receipt = Receipt.from(OrderDto(0))
    }
}
